### ISSUE TITLE
JsonIgnoreProperties annotations are located at the wrong place

### ISSUE THREAD

**Issue by Tcharl** [jhipster/generator-jhipster#7611]
##### **Overview of the issue**

From what I understand, the JsonIgnoreProperties are generated for elasticsearch (to avoid to persist children). But they are generated with the wrong configuration.

Sample of a generated code:
```
public class ConfigurerComponent implements Serializable {
@ManyToOne
    @JsonIgnoreProperties("components")
    private ITSWidget widget;
}
public class ITSWidget implements Serializable {
@OneToMany(mappedBy = "widget")
    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
    private Set<ConfigurerComponent> components = new HashSet<>();
}
```

IMHO, the code should be as follow:
```
public class ConfigurerComponent implements Serializable {
@ManyToOne
    private ITSWidget widget;
}
public class ITSWidget implements Serializable {
@OneToMany(mappedBy = "widget")
    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
    @JsonIgnore
    private Set<ConfigurerComponent> components = new HashSet<>();
}
```

Or may be I'm wrong.

##### **Motivation for or Use Case**

I think that actually, that annotation is ignored as it does nothing.

##### **Reproduce the error**

Sample jdl:
```

entity ConfigurerComponent {
	id Long
    version Long
	operation String
	timestamp Instant
	componentId String required
}
entity ITSWidget {
	id Long
	version Long
	operation String
	timestamp Instant
	name String required minlength(2)
	url String required
	onlyCurrentSprint Boolean
	filterQuery String
}
relationship OneToMany {
    ITSWidget{components} to ConfigurerComponent{widget}
}
search ITSWidget with elasticsearch except ConfigurerComponent

```
##### **Related issues**

N/A

##### **Suggest a Fix**

I can PR when I have time (but I'm kinda busy for the moment)

##### **JHipster Version(s)**

5.0.0-beta.1

##### **JHipster Version(s)**

```
widget@0.0.0 
├── generator-jhipster@5.0.0-beta.1 
├─┬ generator-jhipster-elasticsearch-reindexer@1.1.0
│ └── generator-jhipster@4.14.3 
└─┬ generator-jhipster-enterprise-pom@0.0.5
  └── generator-jhipster@4.14.3 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "xxx",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.0-beta.1",
    "baseName": "widget",
    "packageName": "xxx",
    "packageFolder": "xxx",
    "serverPort": "8081",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "enableTranslation": true,
    "applicationType": "microservice",
    "testFrameworks": [
      "cucumber"
    ],
    "jhiPrefix": "jhi",
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fr"
    ],
    "otherModules": [
      {
        "name": "generator-jhipster-elasticsearch-reindexer",
        "version": "1.1.0"
      },
      {
        "name": "generator-jhipster-enterprise-pom",
        "version": "0.0.5"
      },
      {
        "name": "generator-jhipster-spring-cloud-stream",
        "version": "0.0.2"
      }
    ],
    "clientPackageManager": "yarn",
    "skipClient": true,
    "skipUserManagement": true,
    "reactive": false
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity ITSWidget (its_widget) {
  version Long,
  operation String,
  timestamp Instant,
  name String required minlength(2),
  url String required,
  onlyCurrentSprint Boolean,
  filterQuery String
}
entity ConfigurerComponent (configurer_component) {
  version Long,
  operation String,
  timestamp Instant,
  componentId String required
}
relationship OneToMany {
  ITSWidget{components} to ConfigurerComponent{widget},
}

dto ITSWidget, ConfigurerComponent with mapstruct
angularSuffix ITSWidget, ConfigurerComponent with domain
microservice ITSWidget, ConfigurerComponent with widget
service ITSWidget, ConfigurerComponent with serviceClass
search ITSWidget with elasticsearch (howdy how, that doesn't generate except for search)

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)

git version 2.17.0

node: v8.11.1

npm: 5.6.0

yeoman: 1.8.5

yarn: 1.5.1

Docker version 18.03.1-ce, build 9ee9f40

docker-compose version 1.21.1, build 5a3f1a3

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

See jdl

##### **Browsers and Operating System**

MacOS

- [ x] Checking this box is mandatory (this is just to show you read everything)

Yes I love it!

**Comment by ruddell** [comment]
`@JsonIgnoreProperties` was added in  https://github.com/jhipster/generator-jhipster/issues/6793 to improve JSON serialization for bidirectional relationships, preventing infinite recursion.  The `@JsonIgnoreProperties` means don't map that field to JSON when mapping the parent entity (otherwise it would cause infinite recursion).

I don't think we added anything to prevent indexing child entities though.  There's another related issue about indexing subobjects https://github.com/jhipster/generator-jhipster/issues/6410
